src/transformers/trainer_utils.py

```diff
@@ -1,6 +1,79 @@
+import logging
+import os
 from typing import Dict, NamedTuple, Optional
 
 import numpy as np
+from tqdm.auto import tqdm
+
+
+logger = logging.getLogger(__name__)
+
+try:
+    import wandb
+
+    wandb.ensure_configured()
+    if wandb.api.api_key is None:
+        _has_wandb = False
+        wandb.termwarn("W&B installed but not logged in.  Run `wandb login` or set the WANDB_API_KEY env variable.")
+    else:
+        _has_wandb = False if os.getenv("WANDB_DISABLED") else True
+except (ImportError, AttributeError):
+    _has_wandb = False
+    logger.info(
+        "You are instantiating a Trainer but W&B is not installed. To use wandb logging, "
+        "run `pip install wandb; wandb login` see https://docs.wandb.com/huggingface."
+    )
+
+
+def is_wandb_available():
+    return _has_wandb
+
+
+def setup_wandb(trainer):
+    """
+    Setup the optional Weights & Biases (`wandb`) integration.
+
+    One can override this method to customize the setup if needed.  Find more information at https://docs.wandb.com/huggingface
+    You can also override the following environment variables:
+
+    Environment:
+        WANDB_WATCH:
+            (Optional, ["gradients", "all", "false"]) "gradients" by default, set to "false" to disable gradient logging
+            or "all" to log gradients and parameters
+        WANDB_PROJECT:
+            (Optional): str - "huggingface" by default, set this to a custom string to store results in a different project
+        WANDB_DISABLED:
+            (Optional): boolean - defaults to false, set to "true" to disable wandb entirely
+    """
+    if trainer.is_world_master():
```

Comment:
Here we need to do one of the following:

* Option 1: add a `TFTrainer.is_world_master` method (I would need help for that)
* Option 2: replace with something like `if trainer.__class__.__name__ == "TFTrainer" or trainer.is_world_master()`

What do you think?

Reply:
No need of a `.is_world_master()` method in TF.

Reply:
Ok I moved it directly to `Trainer` so it does not apply to `TFTrainer`